### PR TITLE
Policy wise grouping feature - java code changes

### PR DESCRIPTION
**Pull request by lahiru823**

### CODE REVIEW

File: components/appmgt/org.wso2.carbon.appmgt.api/pom.xml [hunk 1]
```diff
@@ -24,6 +24,11 @@
     <name>WSO2 Carbon - Api for APP Management</name>
 
     <dependencies>
+        <dependency>
+            <groupId>com.googlecode.json-simple.wso2</groupId>
+            <artifactId>json-simple</artifactId>
+            <version>1.1.wso2v1</version>
```
Comment (rushmin) [inline components/appmgt/org.wso2.carbon.appmgt.api/pom.xml:30]: dependency versions should not be mentioned in the child poms. They should be defined only in the dependencyManagement section in the top level pom.

File: components/appmgt/org.wso2.carbon.appmgt.api/src/main/java/org/wso2/carbon/appmgt/api/APIProvider.java [hunk 2]
```diff
@@ -477,4 +476,18 @@ public boolean deleteApp(APIIdentifier identifier, SSOProvider ssoProvider) thro
...
+
+
+    /**
+     * Get Application wise policy group list
+     *
+     * @param appId : Application Id
+     * @return list of policy groups
+     * @throws AppManagementException on error
+     */
+    public List<EntitlementPolicyGroup> getPolicyGroupListByApplication(Integer appId) throws
```
Comment (rushmin) [inline components/appmgt/org.wso2.carbon.appmgt.api/src/main/java/org/wso2/carbon/appmgt/api/APIProvider.java:488]: No need to use the Integer object type as the parameter, just int is enough. Java provides unboxing.

File: components/appmgt/org.wso2.carbon.appmgt.api/pom.xml (hunk 1 above)
Comment (lahiru823) [inline components/appmgt/org.wso2.carbon.appmgt.api/pom.xml:30]: Already fixed with a sub sequent commit

File: components/appmgt/org.wso2.carbon.appmgt.api/src/main/java/org/wso2/carbon/appmgt/api/APIProvider.java (hunk 2 above)
Comment (lahiru823) [inline components/appmgt/org.wso2.carbon.appmgt.api/src/main/java/org/wso2/carbon/appmgt/api/APIProvider.java:488]: Already fixed with a sub sequent commit

**Pull request merged**